designs/wahid/src/frontfacing.js

```diff
@@ -29,19 +27,43 @@ export default (part) => {
     paths.seam.line(points.closureBottom).line(points.dartHemLeft)
   }
   paths.seam.close().attr('class', 'fabric')
-
   if (complete) {
-    if (sa) paths.sa = paths.seam.offset(sa).attr('class', 'fabric sa')
+    // Grainline
+    let grainlineVariableLength = points.cfHips.dist(points.closureTop)
+    if (options.hemStyle === 'classic') {
+      points.grainlineFromFrontFacing = new Point(points.hemTip.x, points.closureTop.y)
+      points.grainlineToFrontFacing = new Point(points.hemTip.x, points.hemTip.y)
+    }
+    if (options.hemStyle === 'rounded') {
+      points.grainlineFromFrontFacing = new Point(points.roundEnd.x, points.roundEnd.y).shift(
+        90,
+        grainlineVariableLength
+      )
+      points.grainlineToFrontFacing = new Point(points.roundEnd.x, points.roundEnd.y)
+    }
+    if (options.hemStyle === 'square') {
+      points.grainlineFromFrontFacing = new Point(points.cutonfoldTo.x, points.cutonfoldTo.y).shift(
+        90,
+        grainlineVariableLength
+      )
+      points.grainlineToFrontFacing = new Point(points.cutonfoldTo.x, points.cutonfoldTo.y)
+    }
 
+    macro('grainline', {
```

Comment:
We don't need all this, we can re-use the grainline from the front part. (if you change line 5 as indicated above)